aws/resource_aws_emr_instance_group.go

```diff
@@ -85,197 +126,196 @@ func resourceAwsEMRInstanceGroup() *schema.Resource {
+			Timeout:    10 * time.Minute,
+			Delay:      10 * time.Second,
+			MinTimeout: 3 * time.Second,
 		}
-		marker = respGrps.Marker
-	}
 
-	if len(groups) == 0 {
-		return nil, fmt.Errorf("No instance groups found for EMR Cluster (%s)", clusterId)
+		_, err = stateConf.WaitForState()
+		if err != nil {
+			return fmt.Errorf(
+				"Error waiting for instance (%s) to terminate: %s", d.Id(), err)
+		}
 	}
 
-	return groups, nil
-}
+	if d.HasChange("autoscaling_policy") {
+		var autoScalingPolicy *emr.AutoScalingPolicy
 
-func fetchEMRInstanceGroup(conn *emr.EMR, clusterId, groupId string) (*emr.InstanceGroup, error) {
-	groups, err := fetchAllEMRInstanceGroups(conn, clusterId)
-	if err != nil {
-		return nil, err
-	}
+		err := json.Unmarshal([]byte(d.Get("autoscaling_policy").(string)), &autoScalingPolicy)
+		if err != nil {
+			return fmt.Errorf("error parsing EMR Auto Scaling Policy JSON for update: %s", err)
+		}
 
-	var group *emr.InstanceGroup
-	for _, ig := range groups {
-		if groupId == *ig.Id {
-			group = ig
-			break
+		putAutoScalingPolicy := &emr.PutAutoScalingPolicyInput{
+			ClusterId:         aws.String(d.Get("cluster_id").(string)),
+			AutoScalingPolicy: autoScalingPolicy,
+			InstanceGroupId:   aws.String(d.Id()),
 		}
-	}
 
-	if group != nil {
-		return group, nil
+		_, err = conn.PutAutoScalingPolicy(putAutoScalingPolicy)
+		if err != nil {
+			return fmt.Errorf("error updating autoscaling policy for instance group %q: %s", d.Id(), err)
+		}
 	}
 
-	return nil, emrInstanceGroupNotFound
+	return resourceAwsEMRInstanceGroupRead(d, meta)
 }
 
-func resourceAwsEMRInstanceGroupUpdate(d *schema.ResourceData, meta interface{}) error {
+func resourceAwsEMRInstanceGroupDelete(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*AWSClient).emrconn
 
-	log.Printf("[DEBUG] Modify EMR task group")
-	instanceCount := d.Get("instance_count").(int)
-
+	log.Printf("[WARN] AWS EMR Instance Group does not support DELETE; resizing cluster to zero before removing from state")
 	params := &emr.ModifyInstanceGroupsInput{
 		InstanceGroups: []*emr.InstanceGroupModifyConfig{
 			{
 				InstanceGroupId: aws.String(d.Id()),
-				InstanceCount:   aws.Int64(int64(instanceCount)),
+				InstanceCount:   aws.Int64(0),
 			},
 		},
 	}
 
 	_, err := conn.ModifyInstanceGroups(params)
-	if err != nil {
-		return err
-	}
-
-	stateConf := &resource.StateChangeConf{
-		Pending:    []string{"PROVISIONING", "BOOTSTRAPPING", "RESIZING"},
-		Target:     []string{"RUNNING"},
-		Refresh:    instanceGroupStateRefresh(conn, d.Get("cluster_id").(string), d.Id()),
-		Timeout:    10 * time.Minute,
-		Delay:      10 * time.Second,
-		MinTimeout: 3 * time.Second,
-	}
-
-	_, err = stateConf.WaitForState()
-	if err != nil {
-		return fmt.Errorf(
-			"Error waiting for instance (%s) to terminate: %s", d.Id(), err)
-	}
-
-	return resourceAwsEMRInstanceGroupRead(d, meta)
+	return err
```

Comment:
We should return error context here for operators and code maintainers:

```suggestion
	if err != nil {
		return fmt.Errorf("error draining EMR Instance Group (%s): %s", d.Id(), err)
	}
	
	return nil
```